test/operator-framework-e2e/operator_framework_test.go

```diff
@@ -83,904 +34,96 @@ func TestOperatorFramework(t *testing.T) {
+		Expect(c.Create(ctx, catalog)).NotTo(HaveOccurred())
 	})
-	It("should succeed", func() {
-		By("creating a new operator-sdk project")
-		err = sdkInitialize(sdkInfo)
-		Expect(err).ToNot(HaveOccurred())
-
-		By("creating a new api and controller")
-		err = sdkNewAPIAndController(sdkInfo)
-		Expect(err).ToNot(HaveOccurred())
-
-		By("generating CRD manifests")
-		err = sdkGenerateManifests(sdkInfo)
-		Expect(err).ToNot(HaveOccurred())
-
-		By("generating bundle directory using kustomize")
-		// Creates bundle structure for the given bundle versions
-		// Bundle content is same for the different versions of bundle now
-		for _, b := range bundleInfo.bundles {
-			err = kustomizeGenPlainBundleDirectory(sdkInfo, bundleInfo.baseFolderPath, b)
-			Expect(err).ToNot(HaveOccurred())
-		}
-
-		By("building/pushing/kind loading bundle images from bundle directories")
-		for _, b := range bundleInfo.bundles {
-			dockerContext := filepath.Join(bundleInfo.baseFolderPath, b.bInputDir)
-			dockerfilePath := filepath.Join(dockerContext, "plainbundle.Dockerfile")
-			err = buildPushLoadContainer(b.imageRef, dockerfilePath, dockerContext, kindServer, GinkgoWriter)
-			Expect(err).ToNot(HaveOccurred())
-		}
-
-		By("generating catalog directory by forming FBC and dockerfile using custom function")
-		imageRefsBundleVersions := make(map[string]string)
-		for _, b := range bundleInfo.bundles {
-			imageRefsBundleVersions[b.imageRef] = b.bundleVersion
-		}
-		err = genPlainCatalogDirectory(catalogDInfo, imageRefsBundleVersions) // the bundle image references and their respective versions are passed
-		Expect(err).ToNot(HaveOccurred())
-
-		By("building/pushing/kind loading the catalog images")
-		dockerContext := catalogDInfo.baseFolderPath
-		dockerfilePath := filepath.Join(dockerContext, fmt.Sprintf("%s.Dockerfile", catalogDInfo.catalogDir))
-		err = buildPushLoadContainer(catalogDInfo.imageRef, dockerfilePath, dockerContext, kindServer, GinkgoWriter)
-		Expect(err).ToNot(HaveOccurred())
-
-		By("creating a Catalog CR and verifying the creation of respective packages and bundle metadata")
-		operatorCatalog, err = createCatalogCheckResources(operatorCatalog, catalogDInfo)
-		Expect(err).ToNot(HaveOccurred())
-
-		By("creating an operator CR and verifying the operator operations")
-		namespace := fmt.Sprintf("%s-system", sdkInfo.projectName)
-		operator, err = createOperator(ctx, catalogDInfo.operatorName, operatorAction.installVersion)
-		Expect(err).ToNot(HaveOccurred())
-		checkOperatorOperationsSuccess(operator, catalogDInfo.operatorName, operatorAction.installVersion, bundleInfo.baseFolderPath, namespace)
-
-		By("upgrading an operator and verifying the operator operations")
-		operator, err = upgradeOperator(ctx, catalogDInfo.operatorName, operatorAction.upgradeVersion)
-		Expect(err).ToNot(HaveOccurred())
-		checkOperatorOperationsSuccess(operator, catalogDInfo.operatorName, operatorAction.upgradeVersion, bundleInfo.baseFolderPath, namespace)
```

Comment:
It looks like update to a new version of the operator we built is missing from the new test.
Same for operator deletion below.

What is our position on this? Leave it to the regular e2e to test?

Reply:
I had this as an open question in the PR description:

>The previous implementation included "upgrading" an operator by changing its .spec.version to be the "new" version. This implementation does not have that and instead focuses on the coast-to-coast of build, package, deploy. Is it necessary to include the "upgrading" of an operator as part of this e2e test?

My thinking here is that we should have this test focus solely on the coast-to-coast of build, package, deploy with OLMv1. It seems unnecessary to test the deletion and manual upgrade by modifying the version. If we want to include the coast-to-coast of build, package, deploy, build+package new version, auto-upgrade I think that is reasonable but would prefer that be done in a follow-up to keep the scope of this PR to the simplification of this test.

Reply:
I also don't think there is anything wrong with adding _additional_ coast-to-coast e2e tests as those workflows are enabled

Reply:
Ah, sorry. I missed that. That makes sense.